NPM version: add explicit version configuration

asi345 opened the pull request:
wasm-pack was creating the NPM package from Rust, by adopting the version from Cargo.toml file. However, both libraries should have their own version and changelogs. Now, a specific NPM version is written into NPM_VERSION file, and jq reads this version when `make wasm` is run. Then this version is added to `package.json` of the build.

Right now, NPM version is set to 0.2.1, which was the last version that we already have.

benma commented on NPM_VERSION:
```diff
@@ -0,0 +1 @@
+10.2.0
```
10.2.0? what a big jump :joy: 

please put it back to v0.2.1, the current version.

asi345 replied:
Sorry 😅

benma commented on Makefile:
```diff
@@ -20,6 +20,8 @@ wasm:
 	wasm-pack build --release --features=wasm
 	cp webhid.js pkg/
 	jq '.files += ["webhid.js"]' pkg/package.json > tmp.json && mv tmp.json pkg/package.json
+	$(eval npm_version=$(shell cat NPM_VERSION))
```
why `eval`? can't you just do `npm_version=$(cat NPM_VERSION)`?

what happens if NPM_VERSION contains a newline (code editors often insert a newline at the end)? if it fails in that case it would be good to fix that

asi345 replied:
That does not work as intended, `npm_version` is not set when I assign like that. This was the only solution I could find to make it work. I think it was because of the environment of the makefile, that is different from a shell.
I checked out [this link](https://superuser.com/questions/1752412/variables-not-being-set-in-makefile) to come up with a solution.

asi345 replied:
For new line, I can add one more command to strip the newline character in the end.

benma replied:
if you put `npm_version = $(shell cat NPM_VERSION)` at the top of the Makefile, `eval` is not needed. this is probably better as `eval` is always a bit sketchy :smile:

benma replied:
about newline: it seems to work as is - maybe `jq` already strips it

asi345 replied:
Okay, I do it like that!

benma commented:
Could you also please document this in .github/README.md?

asi345 commented:
Added the documentation and adjusted the version to v0.2.1.

In addition, I tested the case where NPM_VERSION ends with some newline characters. It seems that makefile or shell handles that automatically, so when I print the variable I don't see any newline characters. Therefore, I did not add anymore commands there.

benma reviewed (approved):
tACK, thanks!
